### ISSUE TITLE
[Android] Unable to switch into native context after reset/restart app

### ISSUE THREAD

**Issue by vmaxim** [appium/appium#10046]
## Environment

* Appium version (or git revision) that exhibits the issue: 1.7.2
* Last Appium version that did not exhibit the issue (if applicable): 1.7.1
* Desktop OS/version used to run Appium: Mac|Win
* Node.js version (unless using Appium.app|exe):
* Mobile platform/version under test: Android
* Real device or emulator/simulator: Both
* Appium CLI or Appium.app|exe: CLI

## Details

1. Create new session
2. Reset app
3. Switch to Native context

## Link to Appium logs

```
info HTTP --> POST /wd/hub/session/2a64883f-836c-4b90-b9aa-60f3d3518ec3/context {"name":"NATIVE_APP"}
...
dbug AndroidDriver Available contexts: ["NATIVE_APP","WEBVIEW_io.appium.android.apis","WEBVIEW_chrome"]
ERR! MJSONWP Encountered internal error running command: Error: Didn't know how to handle switching to context 'NATIVE_APP'
ERR! MJSONWP     at AndroidUiautomator2Driver.callee$0$0$ (C:\1\appium-uiautomator2-driver\node_modules\appium-android-driver\lib\commands\context.js:78:11)
ERR! MJSONWP     at tryCatch (C:\1\appium-uiautomator2-driver\node_modules\babel-runtime\regenerator\runtime.js:67:40)
ERR! MJSONWP     at GeneratorFunctionPrototype.invoke [as _invoke] (C:\1\appium-uiautomator2-driver\node_modules\babel-runtime\regenerator\runtime.js:315:22)
ERR! MJSONWP     at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (C:\1\appium-uiautomator2-driver\node_modules\babel-runtime\regenerator\runtime.js:100:21)
ERR! MJSONWP     at invoke (C:\1\appium-uiautomator2-driver\node_modules\babel-runtime\regenerator\runtime.js:136:37)
ERR! MJSONWP     at enqueueResult (C:\1\appium-uiautomator2-driver\node_modules\babel-runtime\regenerator\runtime.js:185:17)
ERR! MJSONWP     at Promise.F (C:\1\appium-uiautomator2-driver\node_modules\core-js\library\modules\$.export.js:30:36)
ERR! MJSONWP     at AsyncIterator.enqueue (C:\1\appium-uiautomator2-driver\node_modules\babel-runtime\regenerator\runtime.js:184:12)
ERR! MJSONWP     at AsyncIterator.prototype.(anonymous function) [as next] (C:\1\appium-uiautomator2-driver\node_modules\babel-runtime\regenerator\runtime.js:100:21)
ERR! MJSONWP     at Object.runtime.async (C:\1\appium-uiautomator2-driver\node_modules\babel-runtime\regenerator\runtime.js:209:12)
ERR! MJSONWP     at AndroidUiautomator2Driver.callee$0$0 [as switchContext] (C:\1\appium-uiautomator2-driver\node_modules\appium-android-driver\build\lib\commands\context.js:145:30)
ERR! MJSONWP     at AndroidUiautomator2Driver.callee$0$0$ (C:\1\appium-uiautomator2-driver\node_modules\appium-android-driver\lib\commands\context.js:56:14)
ERR! MJSONWP     at tryCatch (C:\1\appium-uiautomator2-driver\node_modules\babel-runtime\regenerator\runtime.js:67:40)
ERR! MJSONWP     at GeneratorFunctionPrototype.invoke [as _invoke] (C:\1\appium-uiautomator2-driver\node_modules\babel-runtime\regenerator\runtime.js:315:22)
ERR! MJSONWP     at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (C:\1\appium-uiautomator2-driver\node_modules\babel-runtime\regenerator\runtime.js:100:21)
ERR! MJSONWP     at GeneratorFunctionPrototype.invoke (C:\1\appium-uiautomator2-driver\node_modules\babel-runtime\regenerator\runtime.js:136:37)
info HTTP <-- POST /wd/hub/session/2a64883f-836c-4b90-b9aa-60f3d3518ec3/context 500 2863 ms - 230 
```

## Code To Reproduce Issue [ Good To Have ]

appium/appium-uiautomator2-driver#134

**Comment by mykola-mokhnach** [comment]
@imurchie Can you please take a look?

**Comment by vmaxim** [comment]
completely fixed by appium/appium-android-driver/pull/334
@mykola-mokhnach  Please close
